addons/sale_loyalty/models/sale_order.py

```diff
@@ -172,9 +172,14 @@ def _discountable_order(self, reward):
                 product=line.product_id,
                 partner=line.order_partner_id,
             )
-            # To compute the discountable amount we get the subtotal and add
-            # non-fixed tax totals. This way fixed taxes will not be discounted
-            taxes = line.tax_id.filtered(lambda t: t.amount_type != 'fixed')
+            if reward.program_type == 'gift_card':
+                # In case of gift cards the order can be totally paid with the
+                # card balance
+                taxes = line.tax_id
+            else:
+                # To compute the discountable amount we get the subtotal and add
+                # non-fixed tax totals. This way fixed taxes will not be discounted
+                taxes = line.tax_id.filtered(lambda t: t.amount_type != 'fixed')
```

Comment:
Please use the field `is_payment_program` in the program, to also cover eWallet =]